packages/web/src/brandkit/common/Page.tsx

```diff
@@ -0,0 +1,242 @@
+import Sidebar from 'src/brandkit/common/Sidebar'
+import Topbar from 'src/brandkit/common/TopBar'
+import { Cell, GridRow, Spans } from 'src/layout/GridRow'
+import { ScreenProps, ScreenSizes, withScreenSize } from 'src/layout/ScreenSize'
+import Footer from 'src/shared/Footer.3'
+import menu, { hashNav } from 'src/shared/menu-items'
+import { HEADER_HEIGHT } from 'src/shared/Styles'
+import { colors, standardStyles } from 'src/styles'
+
+const ROOT = menu.BRAND.link
+
+const LOGO_PATH = `${ROOT}/logo`
+
+const COLOR_PATH = `${ROOT}/color`
+
+const TYPE_PATH = `${ROOT}/typography`
+
+// const IMAGERY_PATH = `${ROOT}/key-imagery`
+const ICONS_PATH = `${ROOT}/icons`
+
+const PAGES = [
+  {
+    title: 'Introduction',
+    href: ROOT,
+    sections: [],
+  },
+  {
+    title: 'Logo',
+    href: LOGO_PATH,
+
+    sections: [
+      { title: 'Overview', href: `${LOGO_PATH}#${hashNav.brandLogo.overview}` },
+      { title: 'Space and Sizing', href: `${LOGO_PATH}#${hashNav.brandLogo.space}` },
+      { title: 'Backgrounds', href: `${LOGO_PATH}#${hashNav.brandLogo.backgrounds}` },
+    ],
+  },
+  {
+    title: 'Color',
+    href: COLOR_PATH,
+
+    sections: [
+      { title: 'Overview', href: `${COLOR_PATH}#${hashNav.brandColor.overview}` },
+      { title: 'Background Colors', href: `${COLOR_PATH}#${hashNav.brandColor.backgrounds}` },
+    ],
+  },
+  {
+    title: 'Typography',
+    href: TYPE_PATH,
+
+    sections: [
+      { title: 'Overview', href: `${TYPE_PATH}#${hashNav.brandTypography.overview}` },
+      { title: 'Type Scale', href: `${TYPE_PATH}#${hashNav.brandTypography.scale}` },
+    ],
+  },
+  {
+    title: 'Icons',
+    href: ICONS_PATH,
+    sections: [],
+  },
+  // {
+  //   title: 'Key Imagery',
+  //   href: IMAGERY_PATH,
+
+  //   sections: [
+  //     { title: 'Overview', href: `${IMAGERY_PATH}#${hashNav.brandImagery.overview}` },
+  //     { title: 'Illustrations', href: `${IMAGERY_PATH}#${hashNav.brandImagery.illustrations}` },
+  //     { title: 'Abstract Graphics', href: `${IMAGERY_PATH}#${hashNav.brandImagery.graphics}` },
+  //   ],
+  // },
+]
+
+export const ROUTE_TO_TITLE = PAGES.reduce((mapping, page) => {
+  mapping[page.href] = page.title
+  return mapping
+}, {})
+
+interface Section {
+  id: string
+  children: React.ReactNode
+}
+
+interface Props {
+  router: SingletonRouter
+  sections: Section[]
+}
+
+interface State {
+  routeHash: string
+}
+
+class Page extends React.Component<Props & ScreenProps, State> {
+  state: State = {
+    routeHash: '',
+  }
+
+  ratios: Record<string, { id: string; ratio: number; top: number }> = {}
+
+  observer: IntersectionObserver
+
+  pageRef = React.createRef<View>()
+
+  sectionRefs = this.props.sections.reduce((acc, section) => {
+    acc[section.id] = React.createRef<View>()
+    return acc
+  }, {})
+
+  onChangeHash = () => {
+    this.setState({ routeHash: window.location.hash })
+  }
+
+  onIntersection = (entries: IntersectionObserverEntry[]) => {
```

Comment:
It took me a while to understand what was going on here because the functions are more named on "how" something is happening than "what". the purpose of this is to update the window hash as well as the sidebar - so maybe something like updateCurrentSection or something like that rather than me having to figure out what the intersection is about

Reply:
for sure, there are a lot of functions needed for observation intersection to do what it needs to do. 
took me a while to get it figure out too
weill name more semantically